packages/webapp/src/components/Modals/WaterUseCalculatorModal/index.jsx

```diff
@@ -1,32 +1,315 @@
-import React, { FC } from 'react';
+import React, { useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 import ModalComponent from '../ModalComponent/v2';
-import { ReactComponent as Person } from '../../../assets/images/task/Person.svg';
+import { ReactComponent as Calculator } from '../../../assets/images/task/Calculator.svg';
 import styles from '../QuickAssignModal/styles.module.scss';
 import Button from '../../Form/Button';
 import PropTypes from 'prop-types';
+import Unit from '../../Form/Unit';
+import {
+  water_valve_flow_rate,
+  location_area,
+  irrigation_depth,
+  estimated_duration,
+  percentage_location,
+} from '../../../util/convert-units/unit';
+import Checkbox from '../../Form/Checkbox';
+import { Label } from '../../Typography';
+import { useSelector } from 'react-redux';
+import { cropLocationsSelector } from '../../../containers/locationSlice';
+import { useForm } from 'react-hook-form';
 
-export default function WaterUseCalculatorModal({ dismissModal }) {
+const TotalWaterUsage = ({ totalWaterUsage }) => {
   const { t } = useTranslation();
+  const labelStyle = {
+    fontSize: '16px',
+    lineHeight: '20px',
+    color: 'var(--teal900)',
+  };
+  return (
+    <div
+      style={{
+        display: 'flex',
+        justifyContent: 'space-between',
+        padding: '10px',
+        width: '100%',
+        border: `2px solid var(--teal700)`,
+        borderRadius: '4px',
+        color: 'var(--teal700)',
+        marginLeft: 'auto',
+      }}
+    >
+      <Label style={labelStyle}>{t('ADD_TASK.IRRIGATION_VIEW.TOTAL_WATER_USAGE')}</Label>
+      <Label style={labelStyle}>{totalWaterUsage} l</Label>
+    </div>
+  );
+};
+
+const WaterUseVolumeCalculator = ({ system, setTotalWaterUsage, totalWaterUsage, formState }) => {
+  const { t } = useTranslation();
+  const { register, getValues, watch, control, setValue } = formState();
+  const { estimated_flow_rate, estimated_irrigation_duration } = getValues();
+
+  useEffect(() => {
+    if (estimated_flow_rate && estimated_irrigation_duration) {
+      setTotalWaterUsage(() => estimated_flow_rate * estimated_irrigation_duration);
+    }
+  }, [estimated_irrigation_duration, estimated_flow_rate]);
+
+  const FLOW_RATE = 'estimated_flow_rate';
+  const FLOW_RATE_UNIT = 'estimated_flow_rate_unit';
+  const DEFAULT_LOCATION_FLOW_RATE = 'default_location_flow_rate';
+  const ESTIMATED_DURATION = 'estimated_irrigation_duration';
+  const ESTIMATED_DURATION_UNIT = 'estimated_irrigation_duration_unit';
+
+  return (
+    <>
+      <Unit
+        register={register}
+        displayUnitName={FLOW_RATE_UNIT}
+        label={t('ADD_TASK.IRRIGATION_VIEW.ESTIMATED_FLOW_RATE')}
+        hookFormSetValue={setValue}
+        hookFormGetValue={getValues}
+        hookFromWatch={watch}
+        name={FLOW_RATE}
+        unitType={water_valve_flow_rate}
+        max={999999.99}
+        system={system}
+        control={control}
+        onChangeUnitOption={(e) => {
```

Comment:
This is not working correctly. If you select the same units you are already on it will keep dividing. (eg current value: 60 l/h, "change" units to l/h (even though already this value), new value: 1 l/h).